bw_check_commit_hook/__init__.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+# Copyright (c) Blu Wireless Technology Ltd. 2023, All Rights Reserved
+
+import argparse
+import os
+import sys
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+from subprocess import PIPE, check_output, run
+from typing import DefaultDict, Dict, List, Optional, Union
+from pkg_resources import resource_filename
+import yaml
+
+# template used to add a patch-style header to a diff
+PATCH_TEMPLATE = """\
+From: A Non <a.non@bluwireless.com>
+Date: {current_datetime}
+Subject: [PATCH] {commit_msg}
+---
+{diff}
+--
+"""
+
+def _parse_args() -> Dict[str, Union[Path, List[Path]]]:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--ci",
+        action="store_true",
+        help="Indicates that the (manual) hook is being run in CI",
+    )
+    parser.add_argument(
+        "files",
+        type=Path,
+        nargs="*",
+        default=[],
+        help="The relevant files",
+    )
+    default_config_file = Path(resource_filename(__name__, "data/checkpatch.yaml"))
+    parser.add_argument(
+        "--config-file",
+        type=Path,
+        default=default_config_file,
+        help="Path to config file",
+    )
+    return vars(parser.parse_args())
+
+
+def _load_config(config_file: Path) -> Optional[Dict]:
+    mandatory_keys = ["DIR_CONFIGS"]
+    try:
+        with config_file.open("r") as file:
+            config = yaml.safe_load(file)
+        for key in mandatory_keys:
+            if key not in config:
+                print(f"Missing mandatory key {key} in config file")
+                return None
+        return config
+    except Exception as e:
+        print(f"Error loading YAML config file: {e}")
+        return None
```

Comment:
I think I would prefer the `sys.exit(1)` to occur within this function instead of outside it. The `except` clause could instead catch the specific exceptions that are expected to occur for whichever scenarios, and the `if key not in config` clause could throw a `ValueError` or `RuntimeError` to be caught in that `except` clause.

Reply:
I am a bit confused, moving `sys.exit(1)` to this function and specific exception handling seems conflicting. IIUC, you are suggesting something like this.

```
def _load_config(config_file: Path) -> dict | None:
    mandatory_keys = ["DIR_CONFIGS"]
    try:
        with config_file.open("r") as file:
            config = yaml.safe_load(file)
        for key in mandatory_keys:
            if key not in config:
                raise RuntimeError(f"Missing mandatory key {key} in config file")
        return config
    except FileNotFoundError:
        print(f"Config file not found: {config_file}")
        sys.exit(1)
    except yaml.YAMLError as e:
        print(f"Error loading YAML config file: {e}")
        sys.exit(1)
    except RuntimeError as e:
        print(f"Error loading YAML config file: {e}")
        sys.exit(1)
```